service/src/test/java/uk/gov/hmcts/reform/fpl/service/translations/provider/TranslatableFurtherEvidenceProviderTest.java

```diff
@@ -0,0 +1,269 @@
+                    SELECTED_ORDER_ID));
+
+        }
+
+        @Test
+        void testIfCollectionEmpty() {
+            assertThrows(IllegalArgumentException.class,
+                () -> underTest.provideSelectedItemDocument(CaseData.builder()
+                        .furtherEvidenceDocuments(null)
+                        .build(),
+                    SELECTED_ORDER_ID));
+
+        }
+    }
+
+    @Nested
+    class Accept {
+
+        @Test
+        void testIfMatchingInCollection() {
+
+            boolean actual = underTest.accept(CaseData.builder()
+                    .furtherEvidenceDocuments(List.of(element(SELECTED_ORDER_ID,
+                        SupportingEvidenceBundle.builder().build())))
+                    .build(),
+                SELECTED_ORDER_ID);
+
+            assertThat(actual).isTrue();
+
+        }
+
+        @Test
+        void testIfNotMatchingInCollection() {
+            boolean actual = underTest.accept(CaseData.builder()
+                    .furtherEvidenceDocuments(List.of(element(UUID_1, SupportingEvidenceBundle.builder().build())))
+                    .build(),
+                SELECTED_ORDER_ID);
+
+            assertThat(actual).isFalse();
+
+        }
+
+        @Test
+        void testIfCollectionEmpty() {
+            boolean actual = underTest.accept(CaseData.builder()
+                    .furtherEvidenceDocuments(null)
+                    .build(),
+                SELECTED_ORDER_ID);
+
+            assertThat(actual).isFalse();
+        }
+    }
+
+    @Nested
+    class ApplyTranslatedOrder {
+
+        @Test
+        void applyMatchedOrder() {
+
+            when(time.now()).thenReturn(NOW);
+
+            List<Element<SupportingEvidenceBundle>> bundleCollection = new java.util.ArrayList<>();
+            bundleCollection.add(element(SELECTED_ORDER_ID, SupportingEvidenceBundle.builder().build()));
+
+            Map<String, Object> actual = underTest.applyTranslatedOrder(CaseData.builder()
+                    .furtherEvidenceDocuments(bundleCollection)
+                    .build(),
+                DOCUMENT_REFERENCE,
+                SELECTED_ORDER_ID);
+
+            assertThat(actual).isEqualTo(Map.of("furtherEvidenceDocuments",
+                List.of(element(SELECTED_ORDER_ID, SupportingEvidenceBundle.builder()
+                    .translatedDocument(DOCUMENT_REFERENCE)
+                    .translationUploadDateTime(NOW)
+                    .build())),
+                "furtherEvidenceDocumentsNC",
+                List.of(element(SELECTED_ORDER_ID, SupportingEvidenceBundle.builder()
+                    .translatedDocument(DOCUMENT_REFERENCE)
+                    .translationUploadDateTime(NOW)
+                    .build())))
+            );
+
+        }
+
+        @Test
+        void applyMatchedOrderFilterConfidential() {
+
+            when(time.now()).thenReturn(NOW);
+
+            List<Element<SupportingEvidenceBundle>> bundleCollection = new java.util.ArrayList<>();
```

Comment:
doesn't need to be fully qualified import

Reply:
same for other places in this file

Reply:
👍